src/js/msp/MSPHelper.js

```diff
@@ -1913,19 +1928,21 @@ MspHelper.prototype.sendModeRanges = function(onCompleteCallback) {
     function send_next_mode_range() {
 
         var modeRange = MODE_RANGES[modeRangeIndex];
+        var modeRangeExtra = MODE_RANGES_EXTRA[modeRangeIndex];
 
         var buffer = [];
         buffer.push8(modeRangeIndex)
             .push8(modeRange.id)
             .push8(modeRange.auxChannelIndex)
             .push8((modeRange.range.start - 900) / 25)
-            .push8((modeRange.range.end - 900) / 25);
+            .push8((modeRange.range.end - 900) / 25)
+            .push8(modeRangeExtra.modeLogic)
```

Comment:
This should have a version check, and only append the extra parameters if the API version is >= 1.41.

Reply:
You're right. I thought betaflight handled it already, but it does make sense to also handle it here; it would be more robust.